clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2077,6 +2077,21 @@ void Sema::finalizeSYCLDelayedAnalysis(const FunctionDecl *Caller,
   }
 }
 
+bool Sema::isArrayZeroInitialized(VarDecl *VD, bool CheckValueDependent) {
+  assert(getLangOpts().SYCLIsDevice &&
+         "Should only be called during device compilation");
```

Comment:
I don't see anything SYCL-specific in this function. Should we really restrict it for using by other languages/models?